src/test/java/com/squareup/javapoet/MethodSpecTest.java

```diff
@@ -57,8 +74,55 @@
       MethodSpec.methodBuilder("doSomething").addExceptions(null);
       fail();
     } catch (IllegalArgumentException expected) {
-      assertThat(expected.getMessage())
-          .isEqualTo("exceptions == null");
+      assertThat(expected).hasMessage("exceptions == null");
+    }
+  }
+
+  @Target(ElementType.PARAMETER)
+  @interface Nullable {
+  }
+
+  @SuppressWarnings("unused") // Used via mirror API.
+  abstract static class Everything {
+    @Deprecated protected abstract <T extends Runnable & Closeable> Runnable everything(
+        @Nullable String thing, List<? extends T> things) throws IOException, SecurityException;
+  }
+
+  @Test public void overrideEverything() {
+    TypeElement classElement = getElement(Everything.class);
+    ExecutableElement methodElement = getOnlyElement(methodsIn(classElement.getEnclosedElements()));
+
+    MethodSpec method = MethodSpec.overriding(methodElement).build();
+    assertThat(method.toString()).isEqualTo(""
+        + "@java.lang.Override\n"
+        + "protected <T extends java.lang.Runnable & java.io.Closeable> "
+        + "java.lang.Runnable everything(java.lang.String arg0, java.util.List<? extends T> arg1) "
```

Comment:
I'm pretty sure this is just due to our use of compile-testing which references back into the elements of the test code (cc/ @gk5885). In an annotation processor, you'd have the real names.